ctrl_rate : 50
simulation_flag : true
log_flag : false

# constrains
theta_dot_upper_limit : 4.0
angle_upper_limit : 45.0  # deg

# mpcc
horizon : 30
end_gain_rate : 20.0 #ec, el param rate of end point 
w_c : 6.0
w_l : 20.0
gamma : 10.0
w_u : 0.02
w_deltau : 0.0001

# reference line
N_SPLINE : 5000

# mpcc valid when dist < 0.3
mpcc_valid_dist : 0.3

# obs penalty and constrains
obs_penalty_valid : false
Qobs : 0.01
obs_x : -1.3
obs_y : -1.1
obs_x_r : 0.0000003 # 退化成一个点
obs_y_r : 0.0000001

# generate map
generate_pcd_map_flag : false

# dp
enable_dp_flag : true
dp_length_rate : 0.25 # dp_length / 2 * horizon_dist

hover_height : 1.0